### PR TITLE
Turn CVMFS_CACHE_REFCOUNT on by default

### PR DESCRIPTION
**Pull request by vvolkl**

### CODE REVIEW

**Comment by vvolkl** [comment]
@cernvm-bot unittest

File: cvmfs/cache_posix.h [hunk 1]
```diff
@@ -73,7 +73,7 @@ class PosixCacheManager : public CacheManager {
     const std::string &cache_path,
     const bool alien_cache,
     const RenameWorkarounds rename_workaround = kRenameNormal,
-    const bool do_refcount = false);
+    const bool do_refcount = true);
```
Comment (HereThereBeDragons) [inline cvmfs/cache_posix.h:76]: the following lines need the same change:
 https://github.com/cvmfs/cvmfs/blob/64595763e066911ed28332dfb92bfaf7395e72de/cvmfs/cache_posix.h#L142-L143

**Review by HereThereBeDragons — CHANGES_REQUESTED** [review]
aside from the one missing constructor, lgtm

but: we should add in the doc somewhere clearly that refcounted cache manager so far only exists for posix cache manager

File: cvmfs/cache_posix.h (hunk 1 above)
Comment (vvolkl) [reply cvmfs/cache_posix.h:76]: Good catch, thanks!

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

**Pull request merged**